src/__tests__/errorCalculation.js

```diff
@@ -1,10 +1,10 @@
-import errorCalculation from '../errorCalculation';
+import * as errCalc from '../errorCalculation';
 
 function sinFunction([a, b]) {
   return (t) => a * Math.sin(b * t);
 }
 
-describe('errorCalculation test', () => {
+describe('sum of residuals', () => {
   it('Simple case', () => {
```

Comment:
Could we simplify this test by making the `sampleFunction` always evaluate to a constant value (perhaps `0`)? I don't see why it should matter whether we use `sin` or any other particular function since `sumOfResiduals` should be purely a function of the difference between the function's output and the corresponding value in the set of data points. e.g. something like this:

```js
const add = (sum, x) => sum + x;
const addSquares = (sum, x) => sum + x*x;
const sampleFunction = ([k]) => (x => k); // resulting function returns constant
const n = 3;
const xs = new Array(n).fill(0).map((zero, i) => i); // [0, 1, ..., n-1] but doesn't matter because f(x; [k]) -> k
const data = {
  x: xs,
  y: xs.map(sampleFunction([0])) // all zeros
};

expect(errCalc.sumOfResiduals(data, [0], sampleFunction)).toBeCloseTo(0, 1);
expect(errCalc.sumOfResiduals(data, [1], sampleFunction)).toBeCloseTo(n, 1);
expect(errCalc.sumOfResiduals(data, [2], sampleFunction)).toBeCloseTo(2*n, 1);

expect(errCalc.sumOfSquaredResiduals(data, [0], sampleFunction)).toBeCloseTo(0, 1);
expect(errCalc.sumOfSquaredResiduals(data, [1], sampleFunction)).toBeCloseTo(n, 1);
expect(errCalc.sumOfSquaredResiduals(data, [2], sampleFunction)).toBeCloseTo(Math.pow(2*n, 2), 1);
```

Reply:
Yes, a simpler function could be used.

Reply:
I realize now that all your change really does here is update the name. The simplification of the test function is outside the scope of this PR. I've attempted to address this in #26, so after that merges this PR can be rebased to resolve this.